packages/core-app-api/src/apis/implementations/AppLanguageApi/AppLanguageSelector.ts

```diff
@@ -53,49 +58,215 @@ export class AppLanguageSelector implements AppLanguageApi {
       );
     }
 
-    return new AppLanguageSelector(languages, initialLanguage);
+    return new AppLanguageSelector(
+      languages,
+      initialLanguage,
+      options?.storageApi,
+      options?.errorApi,
+    );
   }
 
-  static createWithStorage(options?: AppLanguageSelectorOptions) {
-    const selector = AppLanguageSelector.create(options);
+  static createWithStorage(
+    options: Omit<AppLanguageSelectorOptions, 'storageApi' | 'errorApi'>,
```

Comment:
This just gets a bit wonky. Keep it as a single object argument and add them as required here instead.
`options: AppLanguageSelectorOptions & { storageApi: StorageApi; errorApi: ErrorApi }` for example.